src/hotspot/share/asm/register.hpp

```diff
@@ -245,14 +253,44 @@ inline ReverseRegSetIterator<RegImpl> AbstractRegSet<RegImpl>::rbegin() {
 
 // Debugging support
 
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(AbstractRegSet<R> allocated_regs, R first_register) {
+  return ! allocated_regs.contains(first_register);
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(AbstractRegSet<R> allocated_regs, R first_register, Rx... more_registers) {
+  if (allocated_regs.contains(first_register))
+    return false;
+  return different_registers(allocated_regs + first_register, more_registers...);
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(R first_register, Rx... more_registers) {
+  return different_registers(AbstractRegSet<R>(first_register), more_registers...);
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(R reg1, R reg2) {
+  return reg1 != reg2;
+}
+
+template<typename R, typename... Rx>
+inline constexpr bool different_registers(R reg1, R reg2, R reg3) {
+  return reg1 != reg2 && reg2 != reg3 && reg1 != reg3;
+}
```

Comment:
The 2-R and 3-R overloads are just an optimization (probably) of the variadic-R overload for small numbers
of arguments.  Given this is debug-only code, is it really worth the additional source code?